chain/src/store.rs

```diff
@@ -115,16 +115,20 @@ impl ChainStore {
 
 	/// Get PMMR pos for the given output commitment.
 	pub fn get_output_pos(&self, commit: &Commitment) -> Result<u64, Error> {
-		self.get_output_pos_height(commit).map(|(pos, _)| pos)
+		match self.get_output_pos_height(commit) {
+			Ok(Some((pos, _))) => Ok(pos),
+			Ok(None) => Err(Error::NotFoundErr(format!(
+				"Output position for: {:?}",
+				commit
+			))),
+			Err(e) => Err(e),
+		}
 	}
 
 	/// Get PMMR pos and block height for the given output commitment.
-	pub fn get_output_pos_height(&self, commit: &Commitment) -> Result<(u64, u64), Error> {
-		option_to_not_found(
-			self.db
-				.get_ser(&to_key(OUTPUT_POS_PREFIX, &mut commit.as_ref().to_vec())),
-			|| format!("Output position for: {:?}", commit),
-		)
+	pub fn get_output_pos_height(&self, commit: &Commitment) -> Result<Option<(u64, u64)>, Error> {
```

Comment:
We have `get_output_pos_height()` on the batch and on the store itself. We should keep this fns consistent with each other if possible.

And we call this in a few places in txhashset.rs and I think we then need to account for `None` when we call it here -

https://github.com/mimblewimble/grin/blob/db6fb3a9922b970d7cadc2b87ec1783e6c825b4f/chain/src/txhashset/txhashset.rs#L428

Reply:
Good call, forgot about the batch. Updating it now

Reply:
Having to keep them consistent makes me think we should put those functions in a trait and have both the store and batch implement them. But we can do that later in a new PR :)